sale_delivery_date/models/stock_picking.py

```diff
@@ -22,6 +22,18 @@ class StockPicking(models.Model):
     )
     expected_delivery_date = fields.Datetime(compute="_compute_expected_delivery_date")
 
+    def _create_backorder(self):
+        res = super()._create_backorder()
+        now = fields.Datetime.now()
+        for picking in res:
+            # If the scheduled_date is before the current datetime, then date_deadline
+            # cannot be satisfied. Therefore, we need to recompute move's dates
+            if picking.scheduled_date < now:
```

Comment:
* This should be only for outgoing picking
* Only moves where date < now should be updated as you can have different dates on each moves
* Why computing each move separately? They all belong to the same warehouse and for the same partner, so the result should be the same for all the moves
* most importantly, I don't understand why the date should change on the backorder creation. It could still be processed the same day